component/service/src/test/java/org/exoplatform/social/rest/impl/site/SiteRestTest.java

```diff
@@ -95,18 +124,76 @@ public void testGetSiteNavigation() throws Exception {
+                                                                         "siteUn",
+                                                                         true,
+                                                                         1,
+                                                                         String.valueOf(createContainerInstance().getId()));
+    org.exoplatform.portal.jdbc.entity.SiteEntity site2 = creatSiteEntity(SiteType.PORTAL,
+                                                                         "siteDeux",
+                                                                         true,
+                                                                         1,
+                                                                         String.valueOf(createContainerInstance().getId()));
+    site2.setBannerFileId(1);
+    siteDAO.update(site2);
+    String token = RestUtils.generateAttachmentToken(site.getName(), RestUtils.ATTACHMENT_BANNER_TYPE);
+
+    String path = originPath + "notValidSiteName/banner" + "?r=" + token;
+    ContainerResponse resp = launcher.service("GET", path, "", null, null, envctx);
+    assertEquals(404, resp.getStatus());
+
+    startUserSession(IdentityConstants.ANONIM);
+    path = originPath + site2.getName() + "/banner?r=invalid_token";
+    resp = launcher.service("GET", path, "", null, null, envctx);
+    assertEquals(403, resp.getStatus());
+
+    startUserSession("root1");
+    path = originPath + site.getName() + "/banner?r=" + token;
+    resp = launcher.service("GET", path, "", null, null, envctx);
+    assertEquals(200, resp.getStatus());
+    Object entity = resp.getEntity();
+    assertNotNull(entity);
+  }
+
+  private org.exoplatform.portal.jdbc.entity.SiteEntity creatSiteEntity(SiteType siteType,
+                                                                        String siteName,
+                                                                        boolean displayed,
+                                                                        int displayOrder,
+                                                                        String bodyId) {
     org.exoplatform.portal.jdbc.entity.SiteEntity siteEntity = new org.exoplatform.portal.jdbc.entity.SiteEntity();
     siteEntity.setDescription("testDesc");
     siteEntity.setLabel("testLbl");
     siteEntity.setLocale("testLocale");
     siteEntity.setName(siteName);
-    siteEntity.setSiteBody("[{\"children\":[],\"id\":1,\"type\":\"CONTAINER\"}]");
+    siteEntity.setSiteBody("[{\"children\":[],\"id\":" + bodyId + ",\"type\":\"CONTAINER\"}]");
     siteEntity.setSiteType(siteType);
     siteEntity.setSkin("testSkin");
     siteEntity.setProperties("{}");
     siteEntity.setDisplayed(displayed);
     siteEntity.setDisplayOrder(displayOrder);
     return siteDAO.create(siteEntity);
   }
+
+  private ContainerEntity createContainerInstance() {
```

Comment:
why do we need ContainerEntity ?

Reply:
can you consider this https://github.com/Meeds-io/social/pull/2851/files#r1326083267 ?